src/Symbol.ts

```diff
@@ -0,0 +1,182 @@
+import has from './has';
+import global from './global';
+
+export namespace Shim {
+	export let Symbol: SymbolConstructor;
+	let InternalSymbol: SymbolConstructor;
+
+	export interface Symbol {
+		toString(): string;
+		valueOf(): Object;
+		[Symbol.toStringTag]: string;
+		[Symbol.toPrimitive]: symbol;
+		[s: string]: any;
+	}
+
+	export interface SymbolConstructor {
+		prototype: Symbol;
+		(description?: string|number): symbol;
+		for(key: string): symbol;
+		keyFor(sym: symbol): string;
+		hasInstance: symbol;
+		isConcatSpreadable: symbol;
+		iterator: symbol;
+		match: symbol;
+		replace: symbol;
+		search: symbol;
+		species: symbol;
+		split: symbol;
+		toPrimitive: symbol;
+		toStringTag: symbol;
+		unscopables: symbol;
+	}
+
+	const defineProperties = Object.defineProperties;
+	const defineProperty = Object.defineProperty;
+	const create = Object.create;
+
+	const objPrototype = Object.prototype;
+
+	interface GlobalSymbols {
+		[key: string]: symbol;
+	}
+
+	const globalSymbols: GlobalSymbols = {};
+
+	interface TypedPropertyDescriptor<T> extends PropertyDescriptor {
+		value?: T;
+		get? (): T;
+		set? (v: T): void;
+	}
+
+	/**
+	 * Helper function to generate a value property descriptor
+	 * @param {T}                            value        The value the property descriptor should be set to
+	 * @param {boolean}                      enumerable   If the property should be enumberable, defaults to false
+	 * @param {boolean}                      writable     If the property should be writable, defaults to true
+	 * @param {boolean}                      configurable If the property should be configurable, defaults to true
+	 * @returns {TypedPropertyDescriptor<T>}              The property descriptor object
+	 */
+	function getValueDescriptor<T>(value: T, enumerable: boolean = false, writable: boolean = true, configurable: boolean = true): TypedPropertyDescriptor<T> {
+		return {
+			value: value,
+			enumerable: enumerable,
+			writable: writable,
+			configurable: configurable
+		};
+	}
+
+	const getSymbolName = (function () {
+		const created = create(null);
+		return function (desc: string|number): string {
+			let postfix = 0;
+			let name: string;
+			while (created[String(desc) + (postfix || '')]) {
+				++postfix;
+			}
+			desc += String(postfix || '');
+			created[desc] = true;
+			name = '@@' + desc;
+			defineProperty(objPrototype, name, {
```

Comment:
I can understand why it would make sense to add the so-called "well-known" symbols to `Object.prototype`, but what reason is there for decorating `Object.prototype` with *all* shimmed symbols?